File: 415. Add Strings/solution.js
/**
 * @param {string} num1
 * @param {string} num2
 * @return {string}
 */
var addStrings = function(num1, num2) {
  const arr1 = num1.split("");
  const arr2 = num2.split("");
  for (let i = 0; i < arr1.length; i++) arr1[i] = Number(arr1[i]);
  for (let i = 0; i < arr2.length; i++) arr2[i] = Number(arr2[i]);

  let carry = 0;
  let i = arr1.length - 1;
  let j = arr2.length - 1;
  let sum = "";
  while (i >= 0 || j >= 0 || carry == 1) {
    const s1 = i >= 0 ? arr1[i] : 0;
    const s2 = j >= 0 ? arr2[j] : 0;
    sum = ((s1 + s2 + carry) % 10) + sum;
    carry = s1 + s2 + carry > 9 ? 1 : 0;
    i--;
    j--;
  }
  return sum;
};

File: 415. Add Strings/solution.go
func addStrings(num1 string, num2 string) string {
	str1, str2 := []byte(num1), []byte(num2)
	if len(str1) < len(str2) {
		str1, str2 = str2, str1
	}

	sum := byte(0)
	for i, j := len(str1)-1, len(str2)-1; i >= 0 || j >= 0; i, sum = i-1, sum/10 {
		if j >= 0 {
			sum += str2[j] - '0'
			j--
		}
		sum += str1[i] - '0'
		str1[i] = sum%10 + '0'
	}

	if (sum) != byte(0) {
		str1 = append([]byte{'1'}, str1...)
	}
	return string(str1)
}